interconnect/circuit.py

```diff
@@ -585,6 +642,9 @@ def features(self) -> List[generator.Generator]:
         sb_widths = list(self.sbs.keys())
         sb_widths.sort()
         sb_features = [self.sbs[width] for width in sb_widths]
+        if self.core is None:
+            assert len(cb_features) == 0
+            return []
```

Comment:
In the case where core is `None`, but there are some `sb`s, then you may need to return `sb_features` instead of `[]`

Reply:
You're right. It was resolved by another patch: https://github.com/StanfordAHA/garnet/blob/005b5da4420604427b2ea305ed703e4d83c4bbdb/interconnect/circuit.py#L454-L468

See #185.

Reply:
I see, great